tests/scalers_go/new_relic/new_relic_test.go

```diff
@@ -0,0 +1,270 @@
+metadata:
+  labels:
+    name: {{.ServiceName}}
+    namespace: {{.TestNamespace}}
+  annotations:
+    prometheus.io/scrape: "true"
+  name: {{.ServiceName}}
+spec:
+  ports:
+  - name: http
+    port: 80
+    protocol: TCP
+    targetPort: 8080
+  selector:
+    app: {{.DeploymentName}}
+  `
+
+	scaledObjectTemplate = `apiVersion: keda.sh/v1alpha1
+kind: ScaledObject
+metadata:
+  name: {{.ScaledObjectName}}
+  namespace: {{.TestNamespace}}
+  labels:
+    app: {{.DeploymentName}}
+spec:
+  scaleTargetRef:
+    name: {{.DeploymentName}}
+  minReplicaCount: {{.MinReplicaCount}}
+  maxReplicaCount: {{.MaxReplicaCount}}
+  pollingInterval: 1
+  cooldownPeriod:  1
+  advanced:
+    horizontalPodAutoscalerConfig:
+      behavior:
+        scaleDown:
+          stabilizationWindowSeconds: 10
+  triggers:
+    - type: new-relic
+      metadata:
+        account: "{{.NewRelicAccountID}}"
+        region: {{.NewRelicRegion}}
+        threshold: "2"
+        nrql: SELECT rate(sum(http_requests_total), 1 SECONDS) FROM Metric where serviceName='{{.ServiceName}}' and namespaceName='{{.TestNamespace}}' since 120 seconds ago
+      authenticationRef:
+        name: {{.TriggerAuthName}}
+`
+
+	lightLoadTemplate = `apiVersion: v1
+kind: Pod
+metadata:
+  name: fake-light-traffic
+  namespace: {{.TestNamespace}}
+spec:
+  containers:
+  - image: busybox
+    name: test
+    command: ["/bin/sh"]
+    args: ["-c", "while true; do wget -O /dev/null -o /dev/null http://{{.ServiceName}}/; sleep 0.5; done"]`
+
+	heavyLoadTemplate = `apiVersion: v1
+kind: Pod
+metadata:
+  name: fake-heavy-traffic
+  namespace: {{.TestNamespace}}
+spec:
+  containers:
+  - image: busybox
+    name: test
+    command: ["/bin/sh"]
+    args: ["-c", "while true; do wget -O /dev/null -o /dev/null http://{{.ServiceName}}/; sleep 0.1; done"]`
+)
+
+func TestNewRelicScaler(t *testing.T) {
+	// setup
+	t.Log("--- setting up ---")
+	require.NotEmpty(t, newRelicApiKey, "NEWRELIC_API_KEY env variable is required for new relic tests")
+	require.NotEmpty(t, newRelicLicenseKey, "NEWRELIC_LICENSE env variable is required for new relic tests")
+	require.NotEmpty(t, newRelicAccountId, "NEWRELIC_ACCOUNT_ID env variable is required for new relic tests")
+	require.NotEmpty(t, newRelicRegion, "NEWRELIC_REGION env variable is required for new relic tests")
+
+	// Create kubernetes resources
+	kc := GetKubernetesClient(t)
+	data, templates := getTemplateData()
+	CreateKubernetesResources(t, kc, testNamespace, data, templates)
+
+	installNewRelic(t)
+
+	assert.True(t, WaitForDeploymentReplicaReadyCount(t, kc, deploymentName, testNamespace, minReplicaCount, 60, 1),
+		"replica count should be %s after a minute", minReplicaCount)
+
+	// test scaling
+	testActivation(t, kc, data)
+	testScaleUp(t, kc, data)
+	testScaleDown(t, kc, data)
+
+	// cleanup
+	DeleteKubernetesResources(t, kc, testNamespace, data, templates)
+}
+
+func testActivation(t *testing.T, kc *kubernetes.Clientset, data templateData) {
```

Comment:
I can see that you are testing activation, but it hasn't been yet implemented for this scaler, see this issue for details https://github.com/kedacore/keda/issues/2800
Are you up for adding this functionality for this scaler? You can consult PRs for other scalers linked in #2800 on how to achieve that.

Reply:
I'd add that to test the activation, you need to scale from 0, so basically this `testActivation` is not testing anything 😄

Reply:
I was confused about that one. Would you be ok if I throw a proxy in front of it? (nginx perhaps).

If the deployment starts at 9, there will be no way to scale it with requests.

Reply:
Maybe we need to scale another different deployment, I mean, we could have the nginx to receive the traffic and scale another different deployment based on nginx traffic. I know that it could sound as useless, but it's a way to test this behaviour even is not a real case (but a real case could be a queue monitored using new relic where you want to scale from/to zero). 
It's the approach we have followed for [Datadog e2e](https://github.com/kedacore/keda/blob/main/tests/scalers_go/datadog/datadog_test.go)